controller/backing_image_controller.go

```diff
@@ -73,35 +73,44 @@ func NewBackingImageController(
 		ds: ds,
 	}
 
-	ds.BackingImageInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+	var err error
+	if _, err = ds.BackingImageInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
 		AddFunc:    bic.enqueueBackingImage,
 		UpdateFunc: func(old, cur interface{}) { bic.enqueueBackingImage(cur) },
 		DeleteFunc: bic.enqueueBackingImage,
-	})
+	}); err != nil {
+		logrus.WithError(err).Fatal("Failed to register event handler for backing image")
```

Comment:
Didn't notice this `Fatal` used here, which will cause the process to exit immediately. We should just return the err back and let the caller handle instead.

Reply:
https://github.com/longhorn/longhorn-manager/pull/2839